crates/oxc_linter/src/lib.rs

```diff
@@ -62,11 +67,23 @@ pub struct Linter {
     options: LintOptions,
     // config: Arc<LintConfig>,
     config: ConfigStore,
+    // TODO(refactor): remove duplication with `config` field when nested config is
+    // standardized, as we do not need to pass both at that point
+    nested_configs: FxHashMap<PathBuf, ConfigStore>,
```

Comment:
Can this somehow be later marked as non empty? 
Currently this works because `self.config` is then used. 
When no configuration file is found, the default configuration should be used.

Reply:
This is a good question, I'm actually less certain now that we'll need to get rid of the `config` field here, as we do still care about the config in the current working directory or the default config because it's what we use for including/excluding files, for example. I'll think on this a bit more as I'm working on this